src/MongoFramework/Infrastructure/Indexing/IndexModelBuilder.cs

```diff
@@ -64,6 +64,13 @@ private static CreateIndexModel<TEntity> CreateIndexModel(IEntityIndex indexDefi
 					builder.Ascending(indexDefinition.Property.FullPath) : builder.Descending(indexDefinition.Property.FullPath);
 			}
 
+			if (indexDefinition.IndexTenant && typeof(IHaveTenantId).IsAssignableFrom(typeof(TEntity)))
+			{
+				var tenantKey = indexDefinition.SortOrder == IndexSortOrder.Ascending ?
+					builder.Ascending("TenantId") : builder.Descending("TenantId");
+				keyModel = builder.Combine(keyModel, tenantKey);
```

Comment:
One thought I have here is whether we should combine the indexes the other way around. [According to the docs](https://docs.mongodb.com/manual/indexes/#compound-index), index order matters (even though I don't think BSON guarantees property order).

`TenantId` would always exist and always be relevant for entities with `IHaveTenantId`. I think it would make sense for that to be the first property.

```suggestion
				keyModel = builder.Combine(tenantKey, keyModel);
```

Reply:
OK, will change